python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py

```diff
@@ -796,20 +797,13 @@ def load(path, model: Model, device=None):
         if model_type == 'BF16Model':
             result = load_bf16_model(path)
             return patch_attrs(result, model)
-        if isinstance(model, Model):
-            # typically for keras Model
-            model = copy.deepcopy(model)
-            checkpoint_path = metadata.get('checkpoint', None)
-            if checkpoint_path:
-                checkpoint_path = path / metadata['checkpoint']
-                model.load_weights(checkpoint_path)
-                return model
-            else:
-                invalidInputError(False, "Key 'checkpoint' must be specified.")
+        checkpoint_path = metadata.get('checkpoint', None)
+        if checkpoint_path:
+            checkpoint_path = path / metadata['checkpoint']
+            model = keras.models.load_model(checkpoint_path)
+            return model
         else:
```

Comment:
may simplify it to `invalidInputError(checkpoint_path is not None, "Key 'checkpoint' must be specified.")`